cluster-autoscaler/provisioningrequest/conditions/conditions.go

```diff
@@ -40,27 +47,24 @@ const (
 	ExpiredMsg = "ProvisioningRequest is expired"
 )
 
-func shouldCapacityBeBooked(pr *provreqwrapper.ProvisioningRequest) bool {
+// ShouldCapacityBeBooked returns whether capacity should be booked.
+func ShouldCapacityBeBooked(pr *provreqwrapper.ProvisioningRequest) bool {
 	if pr.V1Beta1().Spec.ProvisioningClassName != v1beta1.ProvisioningClassCheckCapacity {
 		return false
 	}
-	if pr.Conditions() == nil || len(pr.Conditions()) == 0 {
+	conditions := pr.Conditions()
+	if apimeta.IsStatusConditionTrue(conditions, v1beta1.Failed) || apimeta.IsStatusConditionTrue(conditions, v1beta1.BookingExpired) {
 		return false
```

Comment:
maybe we can simplify this code to:

```
conditions := pr.Conditions()
 if apimeta.IsStatusConditionTrue(conditions, v1beta1.Provisioned) {
		return true
}
return false
```

Reply:
No, we should check BookingExpired condition. If it's present, the capacity should not be booked

Reply:
The states are no exclusive.
https://maelvls.dev/kubernetes-conditions/#conditions-vs-state-machine